pkg/roachprod/roachprod.go

```diff
@@ -2748,6 +2811,77 @@ func Deploy(
 	return nil
 }
 
+var sideEyeEnvToken, _ = os.LookupEnv("SIDE_EYE_API_TOKEN")
+
+// GetSideEyeTokenFromEnv returns the Side-Eye API token from either an
+// environment variable or gcloud secrets. The second return value is false if
+// the key is not found in either place.
+func GetSideEyeTokenFromEnv() (string, bool) {
+	sideEyeToken := sideEyeEnvToken
+	if sideEyeToken == "" {
+		sideEyeToken = install.GetGcloudSideEyeSecret()
+	}
+	if sideEyeToken == "" {
+		return "", false
+	}
+	return sideEyeToken, true
+}
+
+// CaptureSideEyeSnapshot asks the Side-Eye service to take a snapshot of the
+// cockroach processes of the specified cluster/environment. All errors are
+// logged and swallowed. The agents must previously have been installed
+// through StartSideEyeAgents().
+//
+// sideEyeEnv should generally be the cluster name, unless the agents have been
+// explicitly configured to use a different name.
+//
+// If client is specified, it will be used to communicate with the Side-Eye
+// service. If nil, a client is created and initialized based on the API key
+// form the environment; if the key is not found in the environment, the call is
+// a no-op.
+//
+// On success returns <the snapshot URL>, true. On failure returns "", false.
+func CaptureSideEyeSnapshot(
+	ctx context.Context, l *logger.Logger, sideEyeEnv string, client *sideeyeclient.SideEyeClient,
+) (string, bool) {
+	if client == nil {
+		sideEyeToken, ok := GetSideEyeTokenFromEnv()
+		if !ok {
+			l.Printf("Side-Eye token is not configured via SIDE_EYE_API_TOKEN or gcloud secret, skipping snapshot")
+			return "", false
+		}
+
+		var err error
+		client, err = sideeyeclient.NewSideEyeClient(sideeyeclient.WithApiToken(sideEyeToken))
+		if err != nil {
+			l.Errorf("failed to create Side-Eye client: %s", err)
+			return "", false
+		}
+		defer client.Close()
+	}
+
+	// Protect against the snapshot taking too long.
+	snapCtx, cancel := context.WithTimeout(ctx, time.Second*30)
```

Comment:
How long does it take on average? What if it's a large cluster, would 30 seconds suffice then?